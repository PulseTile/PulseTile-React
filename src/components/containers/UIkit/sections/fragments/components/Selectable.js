import React, { Component } from 'react';

export default class Selectable extends Component {

  state = {
    leftItems: [
      { id: 1, text: 'Item 1' },
      { id: 2, text: 'Item 2' },
      { id: 3, text: 'Item 3' },
      { id: 4, text: 'Item 4' },
      { id: 5, text: 'Item 5' },
      { id: 6, text: 'Item 6' },
    ],
    rightItems: [],
    selectedItemId: null,
  };

  /**
   * This function passes item from left block to right when user makes double click on it
   *
   * @param {number} id
   */
  toggleLeftSelectedItem = id => {
    const { leftItems, rightItems } = this.state;
    const newLeftItems = leftItems.filter(item => {
      return item.id !== id;
    });
    const selectedItem = leftItems.filter(item => {
      return item.id === id;
    });
    const newRightItems = rightItems.concat(selectedItem);
    this.setState({
      leftItems: newLeftItems,
      rightItems: newRightItems,
      selectedItemId: null,
   });
  };

  /**
   * This function passes item from right block to left when user makes double click on it
   *
   * @param {number} id
   */
  toggleRightSelectedItem = id => {
    const { leftItems, rightItems } = this.state;
    const newRightItems = rightItems.filter(item => {
      return item.id !== id;
    });
    const selectedItem = rightItems.filter(item => {
      return item.id === id;
    });
    const newLeftItems = leftItems.concat(selectedItem);
    this.setState({
      leftItems: newLeftItems,
      rightItems: newRightItems,
      selectedItemId: null,
    });
  };

  /**
   * This function sets current selected item
   *
   * @param {number} id
   */
  setSelectedItem = id => {
    this.setState({
      selectedItemId: id,
    });
  };

  /**
   * This function returns className to item <div>
   *
   * @param {number} id
   * @return {string}
   */
  getItemClassName = id => {
    const { selectedItemId } = this.state;
    return (selectedItemId === id) ? "selectable-item selected" : "selectable-item";
  };

  /**
   * This function passes item from left block to right one after ">" clicking
   */
  chooseItem = () => {
    const { leftItems, rightItems, selectedItemId } = this.state;
    if (selectedItemId) {
      const newLeftItems = leftItems.filter(item => {
        return item.id !== selectedItemId;
      });
      const selectedItem = leftItems.filter(item => {
        return item.id === selectedItemId;
      });
      const newRightItems = rightItems.concat(selectedItem);
      this.setState({
        leftItems: newLeftItems,
        rightItems: newRightItems,
        selectedItemId: null,
      });
    }
  };

  /**
   * This function passes item from right block to left one after "<" clicking
   */
  cancelItem = () => {
    const { leftItems, rightItems, selectedItemId } = this.state;
    if (selectedItemId) {
      const newRightItems = rightItems.filter(item => {
        return item.id !== selectedItemId;
      });
      const selectedItem = rightItems.filter(item => {
        return item.id === selectedItemId;
      });
      const newLeftItems = leftItems.concat(selectedItem);
      this.setState({
        leftItems: newLeftItems,
        rightItems: newRightItems,
        selectedItemId: null,
      });
    }
  };

  /**
   * This function passes all items from left block to right one after ">>" clicking
   */
  chooseAll = () => {
    const { leftItems, rightItems } = this.state;
    const newRightItems = rightItems.concat(leftItems);
    this.setState({
      leftItems: [],
      rightItems: newRightItems,
      selectedItemId: null,
    });
  };

  /**
   * This function passes all items from right block to left one after "<<" clicking
   */
  cancelAll = () => {
    const { leftItems, rightItems } = this.state;
    const newLeftItems = leftItems.concat(rightItems);
    this.setState({
      leftItems: newLeftItems,
      rightItems: [],
      selectedItemId: null,
    });
  };

  render() {
    const { leftItems, rightItems } = this.state;
    return (
      <div id="selectable" className="ui-section">
        <strong className="ui-title">Selectable</strong>
        <div className="form-group-wrapper">
          <div className="form-group">
            <div className="selectable">
              <div className="selectable-col selectable-from">
                <label className="control-label">Available Items</label>
                <div className="selectable-list">
                  { leftItems.map((item, key) => {
                      const className = this.getItemClassName(item.id);
                      return (
                        <div key={key} className={className} onDoubleClick={() => this.toggleLeftSelectedItem(item.id)} onClick={() => this.setSelectedItem(item.id)}>
                          <span>{ item.text }</span>
                        </div>
                      );
                    })
                  }
                </div>
              </div>
              <div className="selectable-control">
                 <div className="selectable-control-group">
                    <div className="btn btn-inverse btn-success btn-order" onClick={() => this.chooseItem()}>&rsaquo;</div>
                    <div className="btn btn-inverse btn-success btn-order" onClick={() => this.chooseAll()}>&raquo;</div>
                 </div>
                 <div className="selectable-control-group">
                    <div className="btn btn-inverse btn-success btn-order" onClick={() => this.cancelItem()}>&lsaquo;</div>
                    <div className="btn btn-inverse btn-success btn-order" onClick={() => this.cancelAll()}>&laquo;</div>
                 </div>
               </div>
               <div className="selectable-col selectable-to">
                 <label className="control-label">Selected Items</label>
                 <div className="selectable-list">
                   { rightItems.map((item, key) => {
                       const className = this.getItemClassName(item.id);
                       return (
                         <div key={key} className={className} onDoubleClick={() => this.toggleRightSelectedItem(item.id)} onClick={() => this.setSelectedItem(item.id)}>
                           <span>{ item.text }</span>
                         </div>
                       );
                     })
                   }
                 </div>
               </div>
             </div>
           </div>
        </div>
      </div>
    );
  }
};
