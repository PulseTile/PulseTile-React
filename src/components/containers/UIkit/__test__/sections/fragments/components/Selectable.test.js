import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import Selectable from '../../../../sections/fragments/components/Selectable';

Enzyme.configure({ adapter: new Adapter() });

const BUTTONS_NUMBER = 4;
const ITEMS_NUMBER = 6;

describe('Component <Selectable />', () => {
  it('should renders Selectable with props correctly', () => {
    const component = shallow(<Selectable />);
    expect(component).toMatchSnapshot();
    for (let i = 0; i < ITEMS_NUMBER; i++) {
        component.find('.selectable-item').at(i).simulate('click');
        expect(component).toMatchSnapshot();
    }
    for (let i = 0; i < BUTTONS_NUMBER; i++) {
      component.find('.btn.btn-inverse.btn-success.btn-order').at(i).simulate('click');
      expect(component).toMatchSnapshot();
    }
  });
});
