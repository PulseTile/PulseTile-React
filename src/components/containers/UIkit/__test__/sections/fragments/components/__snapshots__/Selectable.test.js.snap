// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Selectable /> should renders Selectable with props correctly 1`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 2`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item selected"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 3`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item selected"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 4`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item selected"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 5`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item selected"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 6`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item selected"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 7`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
            <div
              className="selectable-item selected"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 8`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 9`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 10`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selectable /> should renders Selectable with props correctly 11`] = `
<div
  className="ui-section"
  id="selectable"
>
  <strong
    className="ui-title"
  >
    Selectable
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <div
        className="selectable"
      >
        <div
          className="selectable-col selectable-from"
        >
          <label
            className="control-label"
          >
            Available Items
          </label>
          <div
            className="selectable-list"
          >
            <div
              className="selectable-item"
              key="0"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 6
              </span>
            </div>
            <div
              className="selectable-item"
              key="1"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 1
              </span>
            </div>
            <div
              className="selectable-item"
              key="2"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 2
              </span>
            </div>
            <div
              className="selectable-item"
              key="3"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 3
              </span>
            </div>
            <div
              className="selectable-item"
              key="4"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 4
              </span>
            </div>
            <div
              className="selectable-item"
              key="5"
              onClick={[Function]}
              onDoubleClick={[Function]}
            >
              <span>
                Item 5
              </span>
            </div>
          </div>
        </div>
        <div
          className="selectable-control"
        >
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ›
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              »
            </div>
          </div>
          <div
            className="selectable-control-group"
          >
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              ‹
            </div>
            <div
              className="btn btn-inverse btn-success btn-order"
              onClick={[Function]}
            >
              «
            </div>
          </div>
        </div>
        <div
          className="selectable-col selectable-to"
        >
          <label
            className="control-label"
          >
            Selected Items
          </label>
          <div
            className="selectable-list"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
